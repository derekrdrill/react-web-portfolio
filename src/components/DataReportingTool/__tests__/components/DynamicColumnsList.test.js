import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import {
  DynamicColumnsList,
  getColumnName,
  getFirstGridContainerJustifyLG,
  getFirstGridItemOrderLG,
  getIconButtonChildren,
  getIconButtonColor,
  getSecondGridContainerJustifyLG,
  getSecondGridItemOrderLG,
} from '../../components/DynamicColumnsList';

import AddCircleIcon from '@mui/icons-material/AddCircle';
import RemoveCircleIcon from '@mui/icons-material/RemoveCircle';

const renderer = new ShallowRenderer();

describe('DynamicColumnsList with me tests', () => {
  it('renders correctly', () => {
    renderer.render(<DynamicColumnsList columns={[{ id: 1, columnName: 'Test 1' }]} buttonClick={jest.fn()} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('runs getColumnName correctly', () => {
    expect(getColumnName('TEST')).toEqual('TEST');
    expect(getColumnName(null)).toEqual('');
  });

  it('runs getIconButtonChildren correctly', () => {
    expect(getIconButtonChildren(true)).toEqual(<AddCircleIcon />);
    expect(getIconButtonChildren(false)).toEqual(<RemoveCircleIcon />);
  });

  it('runs getIconButtonColor correctly', () => {
    expect(getIconButtonColor(true)).toEqual('success');
    expect(getIconButtonColor(false)).toEqual('error');
  });

  it('runs getFirstGridContainerJustifyLG correctly', () => {
    expect(getFirstGridContainerJustifyLG(true)).toEqual('flex-start');
    expect(getFirstGridContainerJustifyLG(false)).toEqual('flex-end');
  });

  it('runs getSecondGridContainerJustifyLG correctly', () => {
    expect(getSecondGridContainerJustifyLG(false)).toEqual('flex-start');
    expect(getSecondGridContainerJustifyLG(true)).toEqual('flex-end');
  });

  it('runs getFirstGridItemOrderLG correctly', () => {
    expect(getFirstGridItemOrderLG(true)).toEqual(1);
    expect(getFirstGridItemOrderLG(false)).toEqual(2);
  });

  it('runs getSecondGridItemOrderLG correctly', () => {
    expect(getSecondGridItemOrderLG(false)).toEqual(1);
    expect(getSecondGridItemOrderLG(true)).toEqual(2);
  });
});
