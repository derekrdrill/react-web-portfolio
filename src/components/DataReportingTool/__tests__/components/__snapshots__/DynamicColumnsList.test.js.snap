// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicColumnsList with me tests renders correctly 1`] = `
<styled.div>
  <Styled(Component)>
    <Styled(Component)
      draggable="true"
    >
      <ForwardRef(Grid)
        container={true}
      >
        <ForwardRef(Grid)
          item={true}
          order={
            Object {
              "lg": 2,
              "xs": 1,
            }
          }
          xs={10}
        >
          <ForwardRef(Grid)
            container={true}
            justifyContent={
              Object {
                "lg": "flex-end",
                "xs": "flex-start",
              }
            }
          >
            <ForwardRef(Typography)
              component="h3"
              variant="subtitle1"
            >
              Test 1
            </ForwardRef(Typography)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          order={
            Object {
              "lg": 1,
              "xs": 2,
            }
          }
          xs={2}
        >
          <ForwardRef(Grid)
            container={true}
            justifyContent={
              Object {
                "lg": "flex-start",
                "xs": "flex-end",
              }
            }
          >
            <ForwardRef(IconButton)
              color="error"
              id={1}
              onClick={[Function]}
              variant="contained"
            >
              <Memo />
            </ForwardRef(IconButton)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </Styled(Component)>
  </Styled(Component)>
</styled.div>
`;
